drivers/hmis/app/models/hmis/workflow_definition/validators/workflow_template_validator.rb

```diff
@@ -0,0 +1,56 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+# Validates Workflow Templates - all nodes are reachable, there is a start and (at least one) end node, etc.
+# Does not inherit from ActiveModel::Validator, since we don't want these validations to run as part of lifecycle hooks,
+# because it's convenient to be able to save the workflow template before creating its nodes and flows.
+# Some of these checks could be validators on the Node subclasses (like EndEvent and StartEvent), but everything lives
+# here to keep the validation logic in one place.
+class Hmis::WorkflowDefinition::Validators::WorkflowTemplateValidator
+  def validate(record)
+    validate_start(record)
+    validate_end(record)
+    validate_tasks_and_gateways(record)
+    validate_nodes_reachable(record)
+  end
+
+  private
+
+  def validate_start(record)
+    unless record.nodes.entrypoints.count == 1
```

Comment:
hypothetically a workflow could have multiple entry points but this seems like a good assumption for now